File: backend/app/schemas/post.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class PostCreate(BaseModel):
    goal_id: uuid.UUID
    caption: str | None = Field(None, max_length=1000)


class PostResponse(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    goal_id: uuid.UUID
    image_url: str | None
    caption: str | None
    created_at: datetime
    reaction_fire: int
    reaction_fist: int
    reaction_party: int
    reaction_heart: int
    # Joined fields
    username: str | None = None
    profile_picture_url: str | None = None
    goal_title: str | None = None
    goal_privacy: str | None = None
    streak_count: int | None = None
    post_user_is_subscribed: bool = False

    model_config = {"from_attributes": True}

File: frontendnew/Streakd/Models/Post.swift
import Foundation

struct Post: Codable, Identifiable, Equatable {
    let id: String
    let userId: String
    let goalId: String
    let imageUrl: String?
    let caption: String?
    let createdAt: String?
    var reactionFire: Int
    var reactionFist: Int
    var reactionParty: Int
    var reactionHeart: Int

    // Joined fields from backend feed query
    let username: String?
    let profilePictureUrl: String?
    let goalTitle: String?
    let streakCount: Int?
    let postUserIsSubscribed: Bool?

    enum CodingKeys: String, CodingKey {
        case id, caption, username
        case userId = "user_id"
        case goalId = "goal_id"
        case imageUrl = "image_url"
        case createdAt = "created_at"
        case reactionFire = "reaction_fire"
        case reactionFist = "reaction_fist"
        case reactionParty = "reaction_party"
        case reactionHeart = "reaction_heart"
        case profilePictureUrl = "profile_picture_url"
        case goalTitle = "goal_title"
        case streakCount = "streak_count"
        case postUserIsSubscribed = "post_user_is_subscribed"
    }
}

struct UserReaction: Codable {
    let postId: String
    let reactEmoji: String

    enum CodingKeys: String, CodingKey {
        case postId = "post_id"
        case reactEmoji = "react_emoji"
    }
}
